chia/full_node/full_node.py

```diff
@@ -1372,6 +1382,37 @@ async def peak_post_processing_2(
         await self.server.send_to_all([msg], NodeType.WALLET)
         self._state_changed("new_peak")
 
+        # Update certain prometheus values that we can't update with a known value in real time elsewhere
+        # Skip updating these values (which result in a few DB queries) if the prometheus server is not enabled
+        if self.prometheus.server.server_enabled:
+            compact_block_count = await self.block_store.count_compactified_blocks()
+            if compact_block_count is not None:
+                self.prometheus.compact_blocks.set(compact_block_count)
+
+            uncompactified_block_count = await self.block_store.count_uncompactified_blocks()
+            if uncompactified_block_count is not None:
+                self.prometheus.uncompact_blocks.set(uncompactified_block_count)
+
+            hint_count = await self.hint_store.count_hints()
+            if hint_count is not None:
+                self.prometheus.hint_count.set(hint_count)
+
+            # Figure out current estimated netspace
+            try:
+                older_block_height = max(0, record.height - int(4608))
+                older_block_header_hash = self.blockchain.height_to_hash(uint32(older_block_height))
+                if older_block_header_hash is None:
+                    raise ValueError(f"Older block hash not found for height {older_block_height}")
+                older_block = await self.block_store.get_block_record(older_block_header_hash)
+                if older_block is None:
+                    raise ValueError("Older block not found")
+                estimated_netspace_bytes = estimate_network_space_bytes(record, older_block, self.constants)
+                # Converting to MiB because prometheus won't currently handle numbers large enough to deal in bytes
+                netspace_mib_estimate = estimated_netspace_bytes / 1048576
+                self.prometheus.netspace_mib.set(netspace_mib_estimate)
+            except ValueError as e:
+                self.log.info(f"Unable to determine current netspace for metrics: {e}.")
```

Comment:
This was intended to just catch any issues we ran across when figuring out netspace only for the purposes of metrics - alternatively, we could just check if not None, and then continue inside of the if, but thats going to end up with a few levels of nested ifs to accomplish the same thing - have to handle the "None" case for the return values though, since they're `Optional` and lint, etc will complain otherwise

Reply:
This is updated to use the context manager as suggested